llvm/test/CodeGen/AIE/aie2/end-to-end/Memops.ll

```diff
@@ -0,0 +1,268 @@
+; CHECK-NEXT:    nop // Delay Slot 5
+; CHECK-NEXT:    nop // Delay Slot 4
+; CHECK-NEXT:    nop // Delay Slot 3
+; CHECK-NEXT:    nop // Delay Slot 2
+; CHECK-NEXT:    nop // Delay Slot 1
+entry:
+  tail call void @llvm.memcpy.p0.p0.i32(ptr noundef nonnull align 4 dereferenceable(3) @buffer1, ptr noundef nonnull align 4 dereferenceable(3) @buffer2, i32 3, i1 false)
+  ret void
+}
+
+; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(readwrite, argmem: none, inaccessiblemem: none)
+define dso_local void @lowerMemcpyUsingWordHalfByte() local_unnamed_addr #0 {
+; CHECK-LABEL: lowerMemcpyUsingWordHalfByte:
+; CHECK:         .p2align 4
+; CHECK-NEXT:  // %bb.0: // %entry
+; CHECK-NEXT:    nopa ; nopb ; movxm p1, #(buffer2+8)
+; CHECK-NEXT:    movxm p2, #(buffer1+8)
+; CHECK-NEXT:    lda.s16 r0, [p1], #2; mov p0, p1
+; CHECK-NEXT:    st.s16 r0, [p2], #2; mov p3, p2
+; CHECK-NEXT:    nop
+; CHECK-NEXT:    nop
+; CHECK-NEXT:    nop
+; CHECK-NEXT:    nop
+; CHECK-NEXT:    nop
+; CHECK-NEXT:    nop
+; CHECK-NEXT:    lda.s8 r0, [p1, #0]
+; CHECK-NEXT:    st.s8 r0, [p2, #0]
+; CHECK-NEXT:    nop
+; CHECK-NEXT:    nop
+; CHECK-NEXT:    nop
+; CHECK-NEXT:    nop
+; CHECK-NEXT:    nop
+; CHECK-NEXT:    paddb [p0], #-8
+; CHECK-NEXT:    lda r0, [p0], #4
+; CHECK-NEXT:    lda r1, [p0, #0]
+; CHECK-NEXT:    nop
+; CHECK-NEXT:    nop
+; CHECK-NEXT:    ret lr
+; CHECK-NEXT:    nop // Delay Slot 5
+; CHECK-NEXT:    paddb [p3], #-8 // Delay Slot 4
+; CHECK-NEXT:    st r0, [p3], #4 // Delay Slot 3
+; CHECK-NEXT:    st r1, [p3, #0] // Delay Slot 2
+; CHECK-NEXT:    nop // Delay Slot 1
+entry:
+  tail call void @llvm.memcpy.p0.p0.i32(ptr noundef nonnull align 4 dereferenceable(11) @buffer1, ptr noundef nonnull align 4 dereferenceable(11) @buffer2, i32 11, i1 false)
+  ret void
+}
+
+; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(readwrite, argmem: none, inaccessiblemem: none)
+define dso_local void @lowerMemcpyUsingVector16() local_unnamed_addr #0 {
+; CHECK-LABEL: lowerMemcpyUsingVector16:
+; CHECK:         .p2align 4
+; CHECK-NEXT:  // %bb.0: // %entry
+; CHECK-NEXT:    nopa ; nopb ; movxm p0, #buffer2; nops
+; CHECK-NEXT:    vlda.128 wh0, [p0]
+; CHECK-NEXT:    nop
+; CHECK-NEXT:    nop
+; CHECK-NEXT:    nop
+; CHECK-NEXT:    nop
+; CHECK-NEXT:    ret lr
+; CHECK-NEXT:    nop // Delay Slot 5
+; CHECK-NEXT:    vmov q0, wh0 // Delay Slot 4
+; CHECK-NEXT:    movxm p0, #buffer1 // Delay Slot 3
+; CHECK-NEXT:    st q0, [p0, #0] // Delay Slot 2
+; CHECK-NEXT:    nop // Delay Slot 1
+entry:
+  tail call void @llvm.memcpy.p0.p0.i32(ptr noundef nonnull align 4 dereferenceable(16) @buffer1, ptr noundef nonnull align 4 dereferenceable(16) @buffer2, i32 16, i1 false)
+  ret void
+}
+
+; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(readwrite, argmem: none, inaccessiblemem: none)
+define dso_local void @lowerMemcpyUsingWordVector16() local_unnamed_addr #0 {
```

Comment:
nit: Vector16 suggests the same vector type as the previous one, but uses a wider vector load?

Reply:
Now it is correct, thank you for pointing that.